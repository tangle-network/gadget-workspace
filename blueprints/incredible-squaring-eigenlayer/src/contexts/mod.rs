pub mod aggregator;
pub mod client;
pub mod x_square;
