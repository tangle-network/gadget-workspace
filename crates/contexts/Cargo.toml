[package]
name = "gadget-contexts"
version = "0.1.0"
authors.workspace = true
edition.workspace = true
license.workspace = true
homepage.workspace = true
repository.workspace = true

[dependencies]
async-trait.workspace = true
gadget-std = { workspace = true }
gadget-clients = { workspace = true }
gadget-keystore = { workspace = true, optional = true }

subxt = { workspace = true, optional = true }

[features]
default = ["std", "keystore"]
std = [
    "gadget-std/std",
    "gadget-keystore?/std",
]

evm = ["gadget-clients/evm"]
eigenlayer = ["gadget-clients/eigenlayer"]
networking = ["gadget-clients/networking"]
keystore = ["gadget-keystore"]
tangle = ["dep:subxt", "gadget-clients/tangle"]