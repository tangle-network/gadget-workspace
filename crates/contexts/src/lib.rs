#[cfg(feature = "eigenlayer")]
pub mod eigenlayer;
#[cfg(feature = "evm")]
pub mod instrumented_evm_client;
#[cfg(feature = "keystore")]
pub mod keystore;
#[cfg(feature = "networking")]
pub mod p2p;

pub mod services;
