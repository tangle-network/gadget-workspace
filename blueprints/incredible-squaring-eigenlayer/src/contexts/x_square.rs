use crate::contexts::client::AggregatorClient;
use gadget_config::StdGadgetConfiguration;
use gadget_macros::contexts::KeystoreContext;

#[derive(Clone, KeystoreContext)]
pub struct EigenSquareContext {
    pub client: AggregatorClient,
    #[config]
    pub std_config: StdGadgetConfiguration,
}
