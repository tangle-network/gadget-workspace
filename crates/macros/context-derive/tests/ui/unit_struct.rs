use gadget_sdk::contexts::KeystoreContext;

#[derive(KeystoreContext)]
struct MyContext;

fn main() {}
