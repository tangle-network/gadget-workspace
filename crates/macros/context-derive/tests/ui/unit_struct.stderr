error: Context Extensions traits cannot be derived for unit structs
 --> tests/ui/unit_struct.rs:4:8
  |
4 | struct MyContext;
  |        ^^^^^^^^^
